### PR TITLE
Refactor / Add Project Organization / Integrate Extractor and Prediction Training

### PR DESCRIPTION
**Pull request by BenCarter44**
Organize project into the various subdirectories.

### CODE REVIEW

**Comment by BenRobotics101** [comment]
- [ ] Add [OSL-Repo-Extractor submodule](https://github.com/mcauley-penney/OSL-repo-extractor/tree/main).

File: src/database_init.py
Comment (BenCarter44) [inline src/database_init.py:1]: TODO: Missing index creation
TODO: Change dictionary cache to SQL cache on PR insert

**Review by BenCarter44 — COMMENTED** [review]
Still have a few small tasks remaining (notes to self)

File: src/database_init.py
Comment (BenCarter44) [reply src/database_init.py:1]: It looks like the changes from this commit: [9b0fc2cd5eaadd8e755b7ca8fac36524b63c97fc](https://github.com/RESHAPELab/ART-CoreEngine/pull/6/commits/9b0fc2cd5eaadd8e755b7ca8fac36524b63c97fc) were reverted, I can put these changes back in.

File: program_analyzer.py [hunk 1]
```diff
@@ -2,35 +2,34 @@
 # main.py
 #
 # By Benjamin Carter, Dylan Johnson, and Hunter Jenkins
-#  
+#
 # This program constitutes the main of the CoreEngine
```
Comment (BenCarter44) [inline program_analyzer.py:7]: TODO (Note to self), Change processFiles() to only process by PR. Rename to process_files() too.

File: src/database_init.py
Comment (mcauley-penney) [reply src/database_init.py:1]: I think that's my fault. I thought I caught that

Comment (BenCarter44) [reply src/database_init.py:1]: Oh, I just fixed it actually.. I wrote those more as notes to self

**Comment by BenCarter44** [comment]
Done. Stand-in working from basic Core-Engine, process files, save to `main.db`

**Pull request merged**